lib/metasploit_data_models/base64_serializer.rb

```diff
@@ -27,7 +27,7 @@ class MetasploitDataModels::Base64Serializer
       },
       lambda { |serialized|
         # Support legacy YAML encoding for existing data
-        YAML.load(serialized)
+        YAML.safe_load(serialized, permitted_classes: MetasploitDataModels::YAML::PERMITTED_CLASSES)
```

Comment:
Context:

https://discuss.rubyonrails.org/t/cve-2022-32224-possible-rce-escalation-bug-with-serialized-columns-in-active-record/81017

Previous stack trace:
```
 2) Mdm::WebPage serialized attributes cookie with WEBrick::Cookie persists successfully
     Failure/Error: let(:web_page) { FactoryBot.create(:mdm_web_page, cookie: cookie) }
     
     Psych::DisallowedClass:
       Tried to load unspecified class: WEBrick::Cookie
     # /Users/user/.rvm/gems/ruby-2.6.6/gems/activerecord-6.1.7/lib/active_record/coders/yaml_column.rb:63:in `yaml_load'
     # /Users/user/.rvm/gems/ruby-2.6.6/gems/activerecord-6.1.7/lib/active_record/coders/yaml_column.rb:26:in `load'
     # /Users/user/.rvm/gems/ruby-2.6.6/gems/activerecord-6.1.7/lib/active_record/type/serialized.rb:22:in `deserialize'
     # /Users/user/.rvm/gems/ruby-2.6.6/gems/activemodel-6.1.7/lib/active_model/type/helpers/mutable.rb:8:in `cast'
     # /Users/user/.rvm/gems/ruby-2.6.6/gems/activemodel-6.1.7/lib/active_model/attribute.rb:175:in `type_cast'
     # /Users/user/.rvm/gems/ruby-2.6.6/gems/activemodel-6.1.7/lib/active_model/attribute.rb:43:in `value'
     # /Users/user/.rvm/gems/ruby-2.6.6/gems/activemodel-6.1.7/lib/active_model/attribute_set.rb:42:in `fetch_value'
     # /Users/user/.rvm/gems/ruby-2.6.6/gems/activerecord-6.1.7/lib/active_record/transactions.rb:408:in `block in restore_transaction_record_state'
     # /Users/user/.rvm/gems/ruby-2.6.6/gems/activemodel-6.1.7/lib/active_model/attribute_set.rb:90:in `transform_values'
     # /Users/user/.rvm/gems/ruby-2.6.6/gems/activemodel-6.1.7/lib/active_model/attribute_set.rb:90:in `map'
     # /Users/user/.rvm/gems/ruby-2.6.6/gems/activerecord-6.1.7/lib/active_record/transactions.rb:407:in `restore_transaction_record_state'
     # /Users/user/.rvm/gems/ruby-2.6.6/gems/activerecord-6.1.7/lib/active_record/transactions.rb:334:in `rolledback!'
     # /Users/user/.rvm/gems/ruby-2.6.6/gems/activerecord-6.1.7/lib/active_record/connection_adapters/abstract/transaction.rb:134:in `rollback_records'
     # /Users/user/.rvm/gems/ruby-2.6.6/gems/activerecord-6.1.7/lib/active_record/connection_adapters/abstract/transaction.rb:312:in `block in rollback_transaction'
     # /Users/user/.rvm/gems/ruby-2.6.6/gems/activesupport-6.1.7/lib/active_support/concurrency/load_interlock_aware_monitor.rb:26:in `block (2 levels) in synchronize'
     # /Users/user/.rvm/gems/ruby-2.6.6/gems/activesupport-6.1.7/lib/active_support/concurrency/load_interlock_aware_monitor.rb:25:in `handle_interrupt'
     # /Users/user/.rvm/gems/ruby-2.6.6/gems/activesupport-6.1.7/lib/active_support/concurrency/load_interlock_aware_monitor.rb:25:in `block in synchronize'
     # /Users/user/.rvm/gems/ruby-2.6.6/gems/activesupport-6.1.7/lib/active_support/concurrency/load_interlock_aware_monitor.rb:21:in `handle_interrupt'
     # /Users/user/.rvm/gems/ruby-2.6.6/gems/activesupport-6.1.7/lib/active_support/concurrency/load_interlock_aware_monitor.rb:21:in `synchronize'
     # /Users/user/.rvm/gems/ruby-2.6.6/gems/activerecord-6.1.7/lib/active_record/connection_adapters/abstract/transaction.rb:309:in `rollback_transaction'
     # /Users/user/.rvm/gems/ruby-2.6.6/gems/activerecord-6.1.7/lib/active_record/connection_adapters/abstract/transaction.rb:325:in `rescue in block in within_new_transaction'
     # /Users/user/.rvm/gems/ruby-2.6.6/gems/activerecord-6.1.7/lib/active_record/connection_adapters/abstract/transaction.rb:322:in `block in within_new_transaction'
     # /Users/user/.rvm/gems/ruby-2.6.6/gems/activesupport-6.1.7/lib/active_support/concurrency/load_interlock_aware_monitor.rb:26:in `block (2 levels) in synchronize'
     # /Users/user/.rvm/gems/ruby-2.6.6/gems/activesupport-6.1.7/lib/active_support/concurrency/load_interlock_aware_monitor.rb:25:in `handle_interrupt'
     # /Users/user/.rvm/gems/ruby-2.6.6/gems/activesupport-6.1.7/lib/active_support/concurrency/load_interlock_aware_monitor.rb:25:in `block in synchronize'
     # /Users/user/.rvm/gems/ruby-2.6.6/gems/activesupport-6.1.7/lib/active_support/concurrency/load_interlock_aware_monitor.rb:21:in `handle_interrupt'
     # /Users/user/.rvm/gems/ruby-2.6.6/gems/activesupport-6.1.7/lib/active_support/concurrency/load_interlock_aware_monitor.rb:21:in `synchronize'
     # /Users/user/.rvm/gems/ruby-2.6.6/gems/activerecord-6.1.7/lib/active_record/connection_adapters/abstract/transaction.rb:317:in `within_new_transaction'
     # /Users/user/.rvm/gems/ruby-2.6.6/gems/activerecord-6.1.7/lib/active_record/connection_adapters/abstract/database_statements.rb:320:in `transaction'
     # /Users/user/.rvm/gems/ruby-2.6.6/gems/activerecord-6.1.7/lib/active_record/transactions.rb:350:in `with_transaction_returning_status'
     # /Users/user/.rvm/gems/ruby-2.6.6/gems/activerecord-6.1.7/lib/active_record/transactions.rb:302:in `save!'
     # /Users/user/.rvm/gems/ruby-2.6.6/gems/activerecord-6.1.7/lib/active_record/suppressor.rb:48:in `save!'
     # /Users/user/.rvm/gems/ruby-2.6.6/gems/factory_bot-6.2.1/lib/factory_bot/evaluation.rb:18:in `create'
     # /Users/user/.rvm/gems/ruby-2.6.6/gems/factory_bot-6.2.1/lib/factory_bot/strategy/create.rb:12:in `block in result'
     # /Users/user/.rvm/gems/ruby-2.6.6/gems/factory_bot-6.2.1/lib/factory_bot/strategy/create.rb:9:in `tap'
     # /Users/user/.rvm/gems/ruby-2.6.6/gems/factory_bot-6.2.1/lib/factory_bot/strategy/create.rb:9:in `result'
     # /Users/user/.rvm/gems/ruby-2.6.6/gems/factory_bot-6.2.1/lib/factory_bot/factory.rb:43:in `run'
     # /Users/user/.rvm/gems/ruby-2.6.6/gems/factory_bot-6.2.1/lib/factory_bot/factory_runner.rb:29:in `block in run'
     # /Users/user/.rvm/gems/ruby-2.6.6/gems/activesupport-6.1.7/lib/active_support/notifications.rb:205:in `instrument'
     # /Users/user/.rvm/gems/ruby-2.6.6/gems/factory_bot-6.2.1/lib/factory_bot/factory_runner.rb:28:in `run'
     # /Users/user/.rvm/gems/ruby-2.6.6/gems/factory_bot-6.2.1/lib/factory_bot/strategy_syntax_method_registrar.rb:28:in `block in define_singular_strategy_method'
     # ./spec/app/models/mdm/web_page_spec.rb:12:in `block (4 levels) in <top (required)>'
     # ./spec/app/models/mdm/web_page_spec.rb:47:in `block (6 levels) in <top (required)>'
     # ./spec/app/models/mdm/web_page_spec.rb:47:in `block (5 levels) in <top (required)>'
     # ------------------
     # --- Caused by: ---
     # Psych::DisallowedClass:
     #   Tried to load unspecified class: WEBrick::Cookie
     #   /Users/user/.rvm/gems/ruby-2.6.6/gems/activerecord-6.1.7/lib/active_record/coders/yaml_column.rb:63:in `yaml_load'
```